### PR TITLE
MINOR: Remove stale streams producer retry default docs.

### PR DESCRIPTION
**Pull request by cwildman**
As part of https://github.com/apache/kafka/pull/5425 the streams default override for producer retries was removed. The documentation was not updated to reflect that change.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by cwildman** [comment]
I don't have permission to assign reviewers or apply the `streams` label.
CC: @bbejeck @mjsax

**Review by mjsax — APPROVED** [review]
LGTM.

Only committers can assign reviews and assign labels. Would you mind to do a PR against https://github.com/apache/kafka-site, too, to get this fixed immediately on the webpage, too?

This should be cherry-picked back to `2.1` when we dropped setting producer retries to 10.

**Comment by cwildman** [comment]
Update to kafka-site: https://github.com/apache/kafka-site/pull/213

**Comment by mjsax** [comment]
Call for second review: @guozhangwang @bbejeck @vvcephei @ableegoldman @abbccdda

**Review by ableegoldman — APPROVED** [review]

**Review by bbejeck — APPROVED** [review]

**Comment by bbejeck** [comment]
Merged #6844 into trunk

**Comment by bbejeck** [comment]
cherry-picked to 2.3, 2.2, and 2.1

**Pull request merged**
